codeinterpreterapi/session.py

```diff
@@ -9,24 +10,45 @@
 from langchain.prompts.chat import MessagesPlaceholder
 from langchain.agents import AgentExecutor, BaseSingleActionAgent
 from langchain.memory import ConversationBufferMemory
-
-from codeinterpreterapi.schema import CodeInterpreterResponse, CodeInput, File, UserRequest
+from langchain.memory.chat_message_histories import (
+    FileChatMessageHistory,
+    RedisChatMessageHistory,
+)
+from codeinterpreterapi.schema import (
+    CodeInterpreterResponse,
+    CodeInput,
+    File,
+    UserRequest,
+)
 from codeinterpreterapi.config import settings
 from codeinterpreterapi.chains.functions_agent import OpenAIFunctionsAgent
 from codeinterpreterapi.prompts import code_interpreter_system_message
 from codeinterpreterapi.callbacks import CodeCallbackHandler
 from codeinterpreterapi.chains.modifications_check import get_file_modifications
 from codeinterpreterapi.chains.remove_download_link import remove_download_link
+import uuid
 
 
 class CodeInterpreterSession:
     def __init__(
         self,
+        session_id=None,
         model=None,
         openai_api_key=settings.OPENAI_API_KEY,
         verbose=settings.VERBOSE,
-        tools: list[BaseTool] = None
+        tools: list[BaseTool] = None,
+        history_storage=os.environ['STORAGE_TYPE'],
     ) -> None:
+        if not session_id:
```

Comment:
Codebox already has an id so maybe just a property?

```python
@property
def session_id(self) -> int: return self.codebox.id
```

Reply:
the purpose of this session_id here is that it could be used to load the old memory of previous chat session.

Reply:
I don't know if it possible to restore a codebox with its id